lib/libthr/thread/thr_attr.c

```diff
@@ -375,7 +375,16 @@ _pthread_attr_init(pthread_attr_t *attr)
 	else {
 		/* Initialise the attribute object with the defaults: */
 		memcpy(pattr, &_pthread_attr_default, sizeof(struct pthread_attr));
-
+#ifdef __CHERI_PURE_CAPABILITY__
+		/*
+		 * A new thread will use the same $ddc as the calling thread
+		 * by default. This can be changed using the
+		 * pthread_attr_setcheriddc_np() API.
+		 *
+		 * XXXAR: should there be an alias CHERI_CR_DDC for CHERI_CR_C0?
+		 */
+		pattr->ddc = cheri_getreg(0);
```

Comment:
On 2 Jun 2016, at 16:18, Brooks Davis <notifications@github.com> wrote:
> In lib/libthr/thread/thr_attr.c:
> 
> > +		/*
> > +		 * A new thread will use the same $ddc as the calling thread
> > +		 * by default. This can be changed using the
> > +		 * pthread_attr_setcheriddc_np() API.
> > +		 *
> > +		 * XXXAR: should there be an alias CHERI_CR_DDC for CHERI_CR_C0?
> > +		 */
> > +		pattr->ddc = cheri_getreg(0);
> This could be __builtin_memcap_global_data_get().
> 

If we don't yet have a cheri_getdefault() or cheri_getddc(), we should add one -- but for some reason I thought we did?

Robert